CODE_OF_CONDUCT.rst

```diff
@@ -0,0 +1,154 @@
+make it easier to enrich all of us and the communities in which we participate.
+
+Importantly: as a member of our community, *you are also a steward of these
+values*.  Not all problems need to be resolved via formal processes, and often
+a quick, friendly but clear word on an online forum or in person can help
+resolve a misunderstanding and de-escalate things.
+
+This code applies equally to founders, developers, mentors and new community
+members, in all spaces managed by Xonsh. This includes all repositories in the
+``xonsh`` GitHub organization, the mailing list, all chat rooms (Gitter and
+IRC), and any other forums created by the project team. In addition, violations
+of this code outside these spaces may affect a person's ability to
+participate within them.
+
+.. note:: It is the strong opinion of the Xonsh developers that the maintainers
+          and contributors of ``xontribs`` (Xonsh plugins / contributions) which
+          are not underneath the ``xonsh`` organization should also abide by
+          these rules.
+
+=================
+Expected behavior
+=================
+
+By embracing the following principles, guidelines and actions to follow or
+avoid, you will help us make Xonsh a welcoming and productive community.
+
+1. **Be friendly and patient**.
+
+2. **Be welcoming**. We strive to be a community that welcomes and supports
+   people of all backgrounds and identities. This includes, but is not limited
+   to, members of any race, ethnicity, culture, national origin, color,
+   immigration status, social and economic class, educational level, sex, sexual
+   orientation, gender identity and expression, age, physical appearance, family
+   status, technological or professional choices, academic
+   discipline, religion, mental ability, and physical ability.
+
+3. **Be considerate**. Your work will be used by other people, and you in turn
+   will depend on the work of others. Any decision you take will affect users
+   and colleagues, and you should take those consequences into account when
+   making decisions. Remember that we're a world-wide community. You may be
+   communicating with someone with a different primary language or cultural
+   background.
+
+4. **Be respectful**. Not all of us will agree all the time, but disagreement is
+   no excuse for poor behavior or poor manners. We might all experience some
+   frustration now and then, but we cannot allow that frustration to turn into a
+   personal attack. It’s important to remember that a community where people
+   feel uncomfortable or threatened is not a productive one.
+
+5. **Be careful in the words that you choose**. Be kind to others. Do not insult
+   or put down other community members. Harassment and other exclusionary
+   behavior are not acceptable. This includes, but is not limited to:
+   * Violent threats or violent language directed against another person
+   * Discriminatory jokes and language
```

Comment:
This feels to me to be extremely vague. It's shown recently, that even some established termini technici (e.g. blacklist/whitelist) can fall into this category, at least according to some SJ trolls.

Regarding jokes, well, a joke is a joke. There are good jokes and there are bad jokes. There might, perhaps, be inappropriate jokes. But "discriminatory jokes"? I don't even know what that means.

Reply:
"Discriminatory Jokes" is a legal & regulatory term (that perhaps those of us living or working in English speaking countries are more familiar with). Here are a couple of definitions/examples: [Canada](https://www.canada.ca/en/department-national-defence/services/benefits-military/defence-ethics/scenarios/dep042019.html), [Australia](https://humanrights.gov.au/quick-guide/12057). 

I think "discrimantory jokes" is covered by "discriminatory language", which is a broader category in text-based interactions like we have in this project. I can see there being an argument for reducing this to "discriminatory language", but I do want to acknowledge that the subclass of jokes is something that does exist in many regulated environments.

Reply:
Thanks for the explanation. I didn't know this is a legal term. It's very sad, that people need regulations of this kind. I'm happy to live in one of the most tolerant countries in the world (according to UN).

Reply:
You can say that again. I wish I lived in a more tolerant place!